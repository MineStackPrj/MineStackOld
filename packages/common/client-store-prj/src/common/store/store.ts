/**
 * @file store.ts
 * @author MineStackPrj
 * @copyright Copyright Â© 2021 MineStackPrj All rights reserved.
 */

import { combineReducers, configureStore } from '@reduxjs/toolkit';

import { indexReducer } from '../reducers';
import { userSignInReducer } from '../slices/userSlice';

export const store = configureStore({
  reducer: indexReducer
});

export const userSignInStore = configureStore({
  reducer: combineReducers({
    user: userSignInReducer
  })
});

export type AppState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
