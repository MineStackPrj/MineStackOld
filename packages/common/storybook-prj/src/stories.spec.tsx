/**
 * @file stories.spec.tsx
 * @author MineStackPrj
 * @copyright Copyright Â© 2021 MineStackPrj All rights reserved.
 */

import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';

import { store } from '@stores/store/store';
import initStoryshots from '@storybook/addon-storyshots';
import { addDecorator } from '@storybook/react';

jest.mock('@material-ui/core', () => {
  const materialUI = jest.requireActual('@material-ui/core');
  return {
    ...materialUI,
    // eslint-disable-next-line @typescript-eslint/naming-convention
    Fade: (props: {children?: React.ReactNode}): JSX.Element => {
      return (
        <>
          <div>Fade</div>
          <div>{props.children}</div>
        </>
      );
    },
    // eslint-disable-next-line @typescript-eslint/naming-convention
    Backdrop: (props: {children?: React.ReactNode}): JSX.Element => {
      return (
        <>
          <div>Backdrop</div>
          <div>{props.children}</div>
        </>
      );
    }
  };
});

addDecorator(story => (
  <Provider store={store}><MemoryRouter initialEntries={['/', 'posts']}>{story()}</MemoryRouter></Provider>
));

initStoryshots();
