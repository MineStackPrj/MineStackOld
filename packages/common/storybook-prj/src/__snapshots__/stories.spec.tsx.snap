// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Common/L1/ErrorMessage Default 1`] = `
<div
  className="makeStyles-root-1 undefined"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
  <p
    className="MuiTypography-root makeStyles-errorMessage-2 MuiTypography-body1 MuiTypography-colorSecondary"
  >
    Error Message
  </p>
</div>
`;

exports[`Storyshots Common/L1/SideMenuItem Primary 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root makeStyles-ListItemPadding-3 MuiListItem-gutters MuiListItem-button"
  data-testid="SideMenu/server"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemIcon-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root material-icons MuiIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.1 0-2 .9-2 2v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 6h-4c0 1.62-1.38 3-3 3s-3-1.38-3-3H5V5h14v4zm-4 7h6v3c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2v-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3z"
      />
    </svg>
  </div>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      サーバー
    </span>
  </div>
</div>
`;
